File: src/fr/gstraymond/ui/view/impl/DescriptionView.java
package fr.gstraymond.ui.view.impl;

import android.text.Html;
import android.text.Spanned;
import android.text.TextUtils;
import android.widget.TextView;
import fr.gstraymond.R;
import fr.gstraymond.biz.CastingCostImageGetter;
import fr.gstraymond.search.model.response.Card;
import fr.gstraymond.tools.DescriptionFormatter;
import fr.gstraymond.ui.CastingCostAssetLoader;
import fr.gstraymond.ui.view.CommonDisplayableView;

public class DescriptionView extends CommonDisplayableView {

	private CastingCostImageGetter imageGetter;
	private DescriptionFormatter descFormatter;

	public DescriptionView(CastingCostAssetLoader castingCostAssetLoader) {
		this.imageGetter = new CastingCostImageGetter(castingCostAssetLoader);
		descFormatter = new DescriptionFormatter();
	}

	@Override
	public void setValue(Card card, int position) {
		String desc = descFormatter.format(card);
		Spanned html = Html.fromHtml(desc, imageGetter, null);
		
		TextView view = (TextView) getView();
		view.setText(html);
	}

	@Override
	public boolean display(Card card) {
		return super.display(! TextUtils.isEmpty(card.getDescription()));
	}

	@Override
	public int getId() {
		return R.id.array_adapter_description;
	}
}

File: app/src/main/kotlin/fr/gstraymond/ui/view/impl/DescriptionView.kt
package fr.gstraymond.ui.view.impl

import android.content.Context
import android.text.Html
import android.widget.TextView
import fr.gstraymond.R
import fr.gstraymond.biz.CastingCostImageGetter
import fr.gstraymond.models.search.response.Card
import fr.gstraymond.tools.DescriptionFormatter
import fr.gstraymond.ui.view.CommonDisplayableView

class DescriptionView(context: Context) : CommonDisplayableView<TextView>(R.id.array_adapter_description) {

    private val descFormatter = DescriptionFormatter()
    private val imageGetter = CastingCostImageGetter.small(context)

    override fun setValue(view: TextView, card: Card, position: Int) {
        val desc = descFormatter.format(card, false)
        val html = Html.fromHtml(desc, imageGetter, null)

        view.text = html
    }

    override fun display(view: TextView, card: Card) = display(view, card.description.isNotEmpty())
}
